src/types/line.js

```diff
@@ -101,7 +102,20 @@ LineAnnotation.defaults = {
 	}
 };
 
-function drawLabel(ctx, line) {
+function calculateAutoRotation(line, chartArea) {
+	const {x, y, x2, y2} = line;
+	let cathetusAdjacent, cathetusOpposite;
+	if (line.options.mode === 'horizontal') {
+		cathetusAdjacent = y2 > y ? chartArea.width : -chartArea.width;
```

Comment:
should not need `chartArea` to calculate the angle between 2 points?
Something like (not tested):
```js
return Math.atan2(x - x2, y - y2);
```

Reply:
> 
> 
> should not need `chartArea` to calculate the angle between 2 points?
> Something like (not tested):
> 
> ```js
> return Math.atan2(x - x2, y - y2);
> ```

Let me check

Reply:
Doesn't work but let me take some time to check better

![autoRot1](https://user-images.githubusercontent.com/11741250/97883144-c66d1d80-1d24-11eb-8274-a5303f5a650d.png)

Reply:
probably need to swap the axes: `Math.atan2(y - y2, x - x2);`

Reply:
already done, but it doesn't still work (for horizontal line)

Reply:
Can I do with my method only without chartArea?

Reply:
```javascript
function calculateAutoRotation(line) {
	const {x, y, x2, y2} = line;
	let cathetusAdjacent, cathetusOpposite;
	if (line.options.mode === 'horizontal') {
		cathetusAdjacent = y2 > y ? x2 - x : -(x2 - x);
		cathetusOpposite = Math.abs(y - y2);
	} else {
		cathetusAdjacent = Math.abs(x - x2);
		cathetusOpposite = x2 > x ? y2 - y : -(y2 - y);
	}
	return Math.atan(cathetusOpposite / cathetusAdjacent);
}
```

Reply:
Lines drawn from origin (middle) out:
![image](https://user-images.githubusercontent.com/27971921/97892694-04743c80-1d39-11eb-800b-dcc6b77c795a.png)

https://codepen.io/kurkle/pen/eYzrxEL

Reply:
`vert` and `horz` are using the method `calc1` implemented and they are correct.

Instead `atan2`, by `calc2` method,  doen't work correctl, as far as I see, because the label is upside down. Am I wrong?

Reply:
I think most will agree with you. Though Annotation lines would most likely be going in the directiond on the right side of that image (where all solutions seem ok). I'd expect a line going right to left (backward)  have the label positioned differently compared to lime going left to right. But as said thats unlikely on annotations and more likely be an error in config. And you can always do the rotation manually. So I'm fine with either.

Reply:
Fine! I agree with you. Let me know if anything is missing from your standpoint.
I have just a question: for many months (I don't want to say years) this plugin was stuck, therefore I think they are many missing things, like callbacks on many options.
Maybe, additional improvements (I have got some in my mind) could be postponed after the release of next version of this plugin.
What do you think?

Reply:
I had some reason I'd like to push a release for v2 before going for first beta for v3.
Soonish.